### PR TITLE
Add ECDH support

### PR DESCRIPTION
**Pull request by dr-orlovsky**
Required for Noise handshake.

This is requiring changes from #117

### CODE REVIEW

**Review by FintanH — COMMENTED** [review]
LGTM :)

File: radicle-crypto/src/lib.rs [hunk 1]
```diff
@@ -156,6 +162,18 @@ impl PublicKey {
 #[derive(Clone, Debug, Eq, PartialEq, Hash)]
 pub struct SecretKey(ed25519::SecretKey);
 
+impl PartialOrd for SecretKey {
```
[the remark this answers is not in the document]
Comment (dr-orlovsky) [reply radicle-crypto/src/lib.rs:165]: Nope, since `ed25519::SecretKey` desn't support it

[the remark this answers is not in the document]
Comment (FintanH) [reply radicle-crypto/src/lib.rs:165]: Oh interesting! I guess the `self.0.cmp` [below](https://github.com/radicle-dev/heartwood/pull/118/files#diff-2811150ba1ee4503d43cfff4bd70e6fe556622dd8090160dc1be1c8543af4b28R173) is acting on the `Deref` of the `SecretKey`, that's what was confusing me :sweat_smile:

**Review by cloudhead — APPROVED** [review]

**Comment by cloudhead** [comment]
Can't seem to push changes to this branch (I think it requires a setting on your side).

I've corrected the typo in the last commit message, and I've specified the `git`-based dependency the way we usually do it, via a `[patch]` entry.

Changes are in 26fa16d2b2988117c06fcf184f4d86d4fd8fec53

**Pull request closed without merging**